packages/services/api/src/modules/organization/providers/organization-members.ts

```diff
@@ -324,10 +310,106 @@ export class OrganizationMembers {
       `,
     );
   }
+
+  static buildOrganizationMembership(
+    rawMembership: z.TypeOf<typeof RawOrganizationMembershipModel>,
+    membershipRole: OrganizationMemberRole,
+    organizationOwnerId: string,
+  ): OrganizationMembership {
+    const resources: ResourceAssignmentGroup = rawMembership.assignedResources ?? {
+      mode: '*',
+      projects: [],
+    };
+
+    return {
+      organizationId: rawMembership.organizationId,
+      userId: rawMembership.userId,
+      isOwner: organizationOwnerId === rawMembership.userId,
+      connectedToZendesk: rawMembership.connectedToZendesk,
+      assignedRole: {
+        role: membershipRole,
+        resources,
+        // We have these as a getter statement as they are
+        // only used in the context of authorization, we do not need
+        // to compute when querying a list of organization mambers via the GraphQL API.
+        get authorizationPolicyStatements() {
+          const resolvedResources = resolveResourceAssignment({
+            organizationId: rawMembership.organizationId,
+            projects: resources,
+          });
+
+          return translateResolvedResourcesToAuthorizationPolicyStatements(
+            rawMembership.organizationId,
+            membershipRole.permissions,
+            resolvedResources,
+          );
+        },
+      },
+      createdAt: rawMembership.createdAt,
+    };
+  }
+
+  static findOrganizationMembership(deps: { logger: Logger; pool: CommonQueryMethods }) {
```

Comment:
I added a static version for looking up a single membership in order to avoid having to instantiate a global singleton instance of `OrganizationMembers`.